### PR TITLE
Selenium: test helper docs

### PR DESCRIPTION
**Pull request by pandafulmanda**
some [docs](https://github.com/openstax/tutor-js/tree/update/test-helper-alias-rename/test-integration/helpers) and [auto-generated docs](http://openstax.github.io/tutor-js/docs/) here

related to #931 and #951 

@openstax/tutor-fe, @openstax/test -- [preview docs here](https://github.com/openstax/tutor-js/tree/update/test-helper-alias-rename/test-integration/helpers) and please let me know your thoughts!

Basically done i think? but awaiting some decisions from #1000  and some updates will need to be made after #992 is merged

### CODE REVIEW

File: test-integration/helpers/README.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    @courseSelect = new CourseSelect(@)
+
+    @user.login(TEACHER_USERNAME)
+
+  @it 'Shows stats for all published plans (readonly)', ->
+    COURSE_CATEGORY = 'BIOLOGY'
+    @courseSelect.goTo(COURSE_CATEGORY)
+    @calendar.waitUntilLoaded()
+
+    @calendar.el.publishedPlan.forEach (plan, index, total) =>
```
Comment (philschatz) [inline test-integration/helpers/README.md]: It seems that for writing the integration tests in testrails `.forEach` is not really used.

Could it be moved into an "advanced" section?

File: test-integration/helpers/README.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    @user = new User(@)
+    @calendar = new CalendarHelper(@)
+    @courseSelect = new CourseSelect(@)
+
+    @user.login(TEACHER_USERNAME)
+
+  @it 'Shows stats for all published plans (readonly)', ->
+    COURSE_CATEGORY = 'BIOLOGY'
+    @courseSelect.goTo(COURSE_CATEGORY)
+    @calendar.waitUntilLoaded()
```
Comment (philschatz) [inline test-integration/helpers/README.md]: Any reason `.goTo` does not already wait until loaded? It seems like adding `.waitUntilLoaded()` everywhere might get forgotten?

File: test-integration/helpers/README.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    @user.goHome()
+```
+
+# Helper Details
+
+From doing the above, if we instantiate CalendarHelper, we get back something that looks like this:
+
+```coffee
+calendarHelper =
+  test: testContext
```
Comment (philschatz) [inline test-integration/helpers/README.md]: What about keeping the `README` to be examples of how to use the helpers and then editing them could be in a `./advanced.md` file?

For example, rather than describe the object this way (which is not easily copy/pasteable & exposes some implementation details), what about something like:

```js
calendar = new helpers.Calendar(this)

calendar.goForecast()
calendar.goStudentScores()
calendar.goPlanByTitle(title)
// This function will block all selenium based promises until
// the loading element is absent before continuing.
calendar.waitUntilLoaded(optionalWaitTime)

// These are elements you can use to test things about the calendar page.
// They are defined by the COMMON_ELEMENTS section in `helpers/calendar.coffee`
calendar.el.forecastLink      // <TestItemHelper>
calendar.el.studentScoresLink // <TestItemHelper>
calendar.el.addToggle         // <TestItemHelper>
```

I think the rest of the items might be too-overwhelming to get an idea of how to use the helpers (maybe a "For creating/editing helpers see `[./advanced.md](./advanced.md)`" link?)

```js
// "The rest of the items":

calendar.test // testContext
calendar.options // options set when the calendar was instantiated

// uses the loadingLocator option
calendar.el.loadingState      // <TestItemHelper>
// from explicitly calling setCommonHelper
calendar.el.planPopup         // <PlanPopupHelper>

// The following are used to attach the helpers that are now on CalendarHelper.el
// They are exposed here for customizing your helper as needed,
// as in this case with the planPopup.
// Usually, you can just rely on the commonElements object.
calendar.setCommonHelper(name, helper)
calendar.setCommonElement(locator, name)
```

**Comment by philschatz** [comment]
Great job documenting how the helpers are structured! I was wondering if the `README.md` might be trying to satisfy 2 audiences:

- using the helpers
- writing helpers

Would it make sense to split the docs up (`README.md` for how to use and a separate file for how to create/edit one of these)?

**Comment by pandafulmanda** [comment]
yes, that makes a lot of sense.  I was struggling with balancing the amount of info vs. what is useful to who. will do that soon

**Comment by philschatz** [comment]
What is this blocked on?

**Comment by pandafulmanda** [comment]
blocked because i think #1003, and #1000 should get merged first, and then I can make edits to this based on the results from those as needed.

**Comment by philschatz** [comment]
Now that #1000 is merged I think this is unblocked, right?
